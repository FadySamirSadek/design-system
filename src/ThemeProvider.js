import React, { Fragment } from 'react'
import styled, { ThemeProvider as Root, injectGlobal } from 'styled-components'
import theme from './theme'

injectGlobal`
* { box-sizing: border-box; }
body {
  margin: 0;
  font-family: ${theme.font};
  line-height: 1.5;
}

@font-face {
  font-family: Averta;
  font-style: normal;
  font-weight: 400;
  src: url(//hackclub.com/fonts/averta-regular.woff2) format('woff2'), url(//hackclub.com/fonts/averta-regular.woff) format('woff');
  unicode-range: U + 0000 - F8FE, U + F900-FFFF;
}
@font-face {
  font-family: Averta;
  font-style: normal;
  font-weight: 700;
  src: url(//hackclub.com/fonts/averta-bold.woff2) format('woff2'), url(//hackclub.com/fonts/averta-bold.woff) format('woff');
  unicode-range: U + 0000 - F8FE, U + F900-FFFF;
}
`

const ThemeProvider = props => (
  <Root
    theme={theme}
    {...props}
    children={<Fragment>{props.children}</Fragment>}
  />
)

export default ThemeProvider
