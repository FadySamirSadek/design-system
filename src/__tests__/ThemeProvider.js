import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider, theme } from '..'
import { Base } from '../ThemeProvider'

describe('ThemeProvider', () => {
  test('renders nothing', () => {
    const json = renderer.create(
      <ThemeProvider>
        <div />
        <div />
        <div />
      </ThemeProvider>
    )
    expect(json).toMatchSnapshot()
  })
})
