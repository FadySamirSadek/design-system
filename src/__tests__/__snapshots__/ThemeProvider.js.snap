// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider renders nothing 1`] = `
Array [
  <div />,
  <div />,
  <div />,
]
`;
